githubapp/client_creator.go

```diff
@@ -236,6 +241,15 @@ func makeUserAgent(base, details string) string {
 	return fmt.Sprintf("%s (%s)", base, details)
 }
 
+func setInstallationIDHeader(installationID int64) ClientMiddleware {
+	return func(next http.RoundTripper) http.RoundTripper {
+		return roundTripperFunc(func(r *http.Request) (*http.Response, error) {
+			r.Header.Set(XInstallationIDHeader, fmt.Sprintf("%d", installationID))
```

Comment:
Looking at this again, it may make more sense to store a value in the request context:

```go
type key string
const installationKey = key("installationID")

// ...

r = r.WithContext(context.WithValue(r.Context(), installationKey, installationID))
return next.RoundTrip(r)
```

That way you avoid the string parsing and also don't send any potentially weird headers to GitHub.